app/public/Model.js

```diff
@@ -22,10 +22,10 @@ export default class Model extends Observable {
         super();
 
         this.session = sessionService.get();
-        this.session.personid = parseInt(this.session.personid, 10); // cast, sessionService has only strings
+        this.session.personid = parseInt(this.session.personid, 10); // Cast, sessionService has only strings
         // TODO if no personid then it is a computer so we need to parse it respectively
-        this.session.roles = this.getRoles();
-        
+        this.session.role = this.getRole();
```

Comment:
If user have permission to more than one detector but not to all detectors, then more than one role is assigned.
So maybe it's better to leave it as: roles instead of role and getRoles() instead of getRole()?
In most cases there will be only one role inside the array but still it's an array of roles not role itself.

Reply:
Ok, I'll change it back